app/services/service.service.js

```diff
@@ -17,37 +16,16 @@ const connectorClient = new ConnectorClient(process.env.CONNECTOR_URL)
  *
  * @returns {Promise<GatewayAccount[]>} promise of collection of gateway accounts which belong to this service
  */
-function getGatewayAccounts (gatewayAccountIds, correlationId) {
-  const accounts = lodash.partition(gatewayAccountIds, id => isADirectDebitAccount(id))
+async function getGatewayAccounts (gatewayAccountIds, correlationId) {
+  const cardGatewayAccountIds = gatewayAccountIds.filter(id => !isADirectDebitAccount(id))
 
-  const fetchCardGatewayAccounts = accounts[1].length > 0
-    ? connectorClient.getAccounts({
-      gatewayAccountIds: accounts[1],
-      correlationId: correlationId
-    }) : Promise.resolve([])
+  const cardGatewayAccounts = await connectorClient.getAccounts({
+    gatewayAccountIds: cardGatewayAccountIds,
+    correlationId: correlationId
+  })
 
-  const fetchDirectDebitGatewayAccounts = accounts[0].length > 0
-    ? directDebitConnectorClient.gatewayAccounts.get({
-      gatewayAccountIds: accounts[0],
-      correlationId: correlationId
-    }) : Promise.resolve([])
-
-  const returnGatewayAccountVariant = ga => (ga.gateway_account_external_id && isADirectDebitAccount(ga.gateway_account_external_id))
-    ? new DirectDebitGatewayAccount(ga).toMinimalJson()
-    : new CardGatewayAccount(ga).toMinimalJson()
-
-  return Promise.all([fetchCardGatewayAccounts, fetchDirectDebitGatewayAccounts])
-    .then(results => {
-      return results
-        .reduce((accumulator, currentValue) => {
-          return currentValue.accounts ? accumulator.concat(currentValue.accounts) : accumulator
-        }, [])
-        .map(returnGatewayAccountVariant)
-        .filter(p => !(p instanceof Error))
-    })
-    .catch(err => {
-      return new Error(err)
-    })
+  return cardGatewayAccounts.accounts
+    .map(gatewayAccount => new CardGatewayAccount(gatewayAccount).toMinimalJson())
```

Comment:
This diff should be used as an example of how to clean up JavaScript, this is so much easier for future devs to pickup and understand. 👍